docs/tutorial/application-architecture.md

```diff
@@ -0,0 +1,141 @@
+
+The main process creates web pages by creating `BrowserWindow` instances. Each
+`BrowserWindow` instance runs the web page in its own renderer process. When a
+`BrowserWindow` instance is destroyed, the corresponding renderer process
+is also terminated.
+
+The main process manages all web pages and their corresponding renderer
+processes. Each renderer process is isolated and only cares about the web page
+running in it.
+
+In web pages, calling native GUI related APIs is not allowed because managing
+native GUI resources in web pages is very dangerous and it is easy to leak
+resources. If you want to perform GUI operations in a web page, the renderer
+process of the web page must communicate with the main process to request that
+the main process perform those operations.
+
+> #### Aside: Communication Between Processes
+> In Electron, we have several ways to communicate between the main process
+and renderer processes. Like [`ipcRenderer`](../api/ipc-renderer.md) and
+[`ipcMain`](../api/ipc-main.md) modules for sending messages, and the
+[remote](../api/remote.md) module for RPC style communication. There is also
+an FAQ entry on [how to share data between web pages][share-data].
+
+## Using Electron APIs
+
+Electron offers a number of APIs that support the development of a desktop
+application in both the main process and the renderer process. In both
+processes, you'd access Electron's APIs by requiring its included module:
+
+```javascript
+const electron = require('electron')
+```
+
+All Electron APIs are assigned a process type. Many of them can only be
+used from the main process, some of them only from a renderer process,
+some from both. The documentation for the individual API will clearly
+state which process they can be used from.
+
+A window in Electron is for instance created using the `BrowserWindow`
+class. It is only available in the main process.
+
+```javascript
+// This will work in the main process, but be `undefined` in a
+// renderer process:
+const { BrowserWindow } = require('electron')
+
+const win = new BrowserWindow()
+```
+
+Since communication between the processes is possible, a renderer process
+can call upon the main process to perform tasks. Electron comes with a
+module called `remote` that exposes APIs usually only available on the
+main process. In order to create a `BrowserWindow` from a renderer process,
+we'd use the remote as a middle-man:
+
+```javascript
+// This will work in a renderer process, but be `undefined` in the
+// main process:
+const { remote } = require('electron')
+const { BrowserWindow } = remote
+
+const win = new BrowserWindow()
+```
+
+## Using Node.js APIs
+
+Electron exposes full access to Node.js both in the main and the renderer
+process. This has two important implications:
+
+1) All APIs available in Node.js are available in Electron. Calling the
+following code from an Electron app works:
+
+```javascript
+const fs = require('fs')
+
+const root = fs.readdirSync('/')
```

Comment:
Does this line work on windows? Home about `os.homedir()` instead?

Reply:
It does! It returns `C:\` by default 👍